src/OpenApi/src/Comparers/OpenApiTagComparer.cs

```diff
@@ -9,7 +9,7 @@ namespace Microsoft.AspNetCore.OpenApi;
 /// This comparer is used to maintain a globally unique list of tags encountered
 /// in a particular OpenAPI document.
 /// </summary>
-internal class OpenApiTagComparer : IEqualityComparer<OpenApiTag>
+internal sealed class OpenApiTagComparer : IEqualityComparer<OpenApiTag>
```

Comment:
This `Equals` seems meaningfully different from the new ones.

Reply:
`OpenApiTags` aren't part of the `OpenApiSchema` type which is the one we care about making equality checks for. This comparer predates this new set and primarily exists to support the top-level tags list that we construct here:

https://github.com/dotnet/aspnetcore/blob/fd73ff293d275b17dfe6afb125cb7ec5f4b859b0/src/OpenApi/src/Services/OpenApiDocumentService.cs#L52-L54